presto-main/src/main/java/com/facebook/presto/event/query/QueryMonitorConfig.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.event.query;
+
+import io.airlift.configuration.Config;
+import io.airlift.units.DataSize;
+import io.airlift.units.DataSize.Unit;
+import io.airlift.units.MaxDataSize;
+import io.airlift.units.MinDataSize;
+
+import javax.validation.constraints.NotNull;
+
+public class QueryMonitorConfig
+{
+    private DataSize maxOutputStageJsonSize = new DataSize(16, Unit.MEGABYTE);
+
+    @MinDataSize("1kB")
+    @MaxDataSize("2147483647B")
+    @NotNull
+    public DataSize getMaxOutputStageJsonSize()
+    {
+        return maxOutputStageJsonSize;
+    }
+
+    @Config("monitor.max-output-stage-json-size")
```

Comment:
I'd call this "event.max-output-stage-size". 

"monitor" is an internal, and rather obscure, implementation detail.